test_plans/kernel_bundle.asciidoc

```diff
@@ -0,0 +1,144 @@
+
+* for kernel with no optional features used and with sycl::requires() kernel attribute
+
+** `true` for `has(fp16)`/`has(fp64)`/`has(atomic64)` and device with appropriate aspect
+** false for `has(fp16)`/`has(fp64)`/`has(atomic64)` and device without appropriate aspect
+
+* for multiple kernels result is equal to AND operation applied to the multiple calls - each one using single kernel id
+** Check with vector with kernel ids for same and other devices
+
+==== Tests for `template<typename KernelName> bool is_compatible(const device &dev)`
+
+Same tests as previous section except no zero, built-in and multiple kernels cases.
+
+=== `sycl::kernel_bundle` API
+
+==== Deleted default constructor
+
+Check that `std::is_default_constructible_v<sycl::kernel_bundle>` is `false`.
+
+==== Test for `get_backend()`
+
+Call function `kernel::get_backend()` and check that return type is `sycl::backend`.
+
+==== Tests for `get_devices()`
+
+* Use some of available devices to create fill std::vector<device>.
+* Create kernel_bundle via `get_kernel_bundle(const context&, const std::vector<device>&)`.
+* Check that return type is std::vector<device>.
+* Check that return vector contains all devices from input vector and only them.
+
+==== `has_kernel(const kernel_id &kernelId, const device &dev)`
+
+Check that result of function is:
+
+* `true` if kernelId is within get_kernel_ids() and `is_compatible({kernelId}, const device&)` is `true`
+* `false` if kernelId is within get_kernel_ids() and `is_compatible({kernelId}, const device&)` is `false`
+* `false` if kernelId is not within get_kernel_ids() and `is_compatible({kernelId}, const device&)` is `true`
+* `false` if kernelId is not within get_kernel_ids() and `is_compatible({kernelId}, const device&)` is `false`
+
+
+==== `std::vector<kernel_id> get_kernel_ids()`
+
+Define 4 kernels.
+Get kernel_bundle with built-in kernels.
+Check that result of get_kernel_ids:
+
+* has type `std::vector<kernel_id>`
+* has size 4
+* has only different kernel ids
+
+==== `get_kernel()`
+
+Define kernel KernelName.
+Get `kernel_bundle` with built-in kernels.
+
+* Use `get_kernel<KernelName>()` to get kernel and that return type is `sycl::kernel`
+* Try to use `get_kernel<OtherKernelName>()` and check that `errc::invalid` is thrown
+
+=== Tests for working with specialization constants
+
+Partially tested in https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/tests/specialization_constants/specialization_constants_via_kernel_bundle.h
+
+There are should be two spec constants defined - `SpecName` and `OtherSpecName`.
+
+==== Empty kernel bundle
+
+* Get an empty `kernel_bundle`.
+* Check that `contains_specialization_constants()` return `false`.
+* Check that `native_specialization_constant()` return `false`.
+* Check that `has_specialization_constant<SpecName>()` return `false`.
+
+==== Kernel bundle without `kernel_handler::get_specialization_constant()` call

```

Comment:
I think this entire test is not valid because the kernel bundle could contain other kernels in the application, and those other kernels might have specialization constants.

Reply:
Removed.